### ISSUE TITLE
[Bug] Descrease performance due to #264

### ISSUE THREAD

**Issue by aolemila** [sgl-project/sglang-jax#291]
### Describe the bug

Descrease performance due to #264

### Reproduction

<img width="630" height="193" alt="Image" src="https://github.com/user-attachments/assets/9d601603-44bb-4a87-9821-9272b0709ad1" />

JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache uv run python -u -m sgl_jax.launch_server --model-path Qwen/Qwen3-8B --trust-remote-code  --tp-size=4 --device=tpu --mem-fraction-static=0.95 --chunked-prefill-size=2048 --download-dir=/tmp --dtype=bfloat16 --max-running-requests 256 --skip-server-warmup --page-size=128  --disable-radix-cache

python3 -m sgl_jax.bench_serving --backend sgl-jax --dataset-name random --num-prompts 48 --random-input 1024 --random-output 1024 --random-range-ratio 1 --warmup-requests 0 --max-concurrency=16

d658235aadfe4f3f889cb44842395ea6c9e71678

<img width="361" height="480" alt="Image" src="https://github.com/user-attachments/assets/b0c82db8-dd61-4722-a5b8-008e55c4c171" />

f072cf502aa7b74615e2df4924e1402d3d6c6120

<img width="381" height="474" alt="Image" src="https://github.com/user-attachments/assets/0600d09e-219d-46c3-9b16-6391a0b3e234" />

### Environment

TPU v6e-4

**Comment by JamesBrianD** [comment]
I will fix it.
